client.go

```diff
@@ -7,11 +7,12 @@ import (
 
 // Client Is a generic HTTP client interface
 type Client interface {
-	Get(url string, headers http.Header) (Response, error)
-	Post(url string, body io.Reader, headers http.Header) (Response, error)
-	Put(url string, body io.Reader, headers http.Header) (Response, error)
-	Patch(url string, body io.Reader, headers http.Header) (Response, error)
-	Delete(url string, headers http.Header) (Response, error)
+	Get(url string, headers http.Header) (*http.Response, error)
+	Post(url string, body io.Reader, headers http.Header) (*http.Response, error)
+	Put(url string, body io.Reader, headers http.Header) (*http.Response, error)
+	Patch(url string, body io.Reader, headers http.Header) (*http.Response, error)
+	Delete(url string, headers http.Header) (*http.Response, error)
+	Do(req *http.Request) (*http.Response, error)
```

Comment:
should we split these ? coz keeping everything would make it hard to extend as embedding structs would've to implement. atleast `Do` could be a separate interface.

we should've a mock for this, so its easy to test.

Reply:
Lets add a issue for this and pick it up as a part of different PR. Its fine to keep the interface same for now